app/models/document.rb

```diff
@@ -157,21 +157,17 @@ def self.all
       fields: [
         :base_path,
         :content_id,
-        :title,
-        :public_updated_at,
-        :details,
-        :description,
       ]
     ).to_ostruct
 
-    # Fetch individual payloads for each `specialist_document`
-    payloads = response.map { |payload| publishing_api.get_content(payload.content_id).to_ostruct }
-
     # Select the ones which match the current document type
-    payloads_of_format = payloads.select { |payload| payload.details.metadata.document_type == self.publishing_api_document_type }
+    payloads_of_format = response.select { |payload| payload.base_path.starts_with?(self.finder_schema.base_path) }
+
+    # Fetch individual payloads for each `specialist_document`
+    payloads = payloads_of_format.map { |payload| publishing_api.get_content(payload.content_id).to_ostruct }
 
     # Deserialize the payloads into real Objects and return them
-    payloads_of_format.map { |payload| self.from_publishing_api(payload) }
+    payloads.map { |payload| self.from_publishing_api(payload) }
```

Comment:
We could've, but it's been discussed before that people are largely unfamiliar with `&method`. I'm very wary of introducing concepts within this App that people aren't familiar with before they've had a chance to get to grips with it.